src/main/kotlin/no/nav/familie/tilbake/api/DokumentController.kt

```diff
@@ -32,15 +33,24 @@ class DokumentController(private val varselbrevService: VarselbrevService,
                          private val henleggelsesbrevService: HenleggelsesbrevService,
                          private val vedtaksbrevService: VedtaksbrevService) {
 
-    @GetMapping("/forhandsvis-manueltVarselbrev/{behandlingId}",
-                produces = [MediaType.APPLICATION_PDF_VALUE])
+    @PostMapping("/forhandsvis/{behandlingId}",
+                 produces = [MediaType.APPLICATION_JSON_VALUE])
     @Rolletilgangssjekk(minimumBehandlerrolle = Behandlerrolle.SAKSBEHANDLER,
                         handling = "Forhåndsviser brev",
                         henteParam = "behandlingId")
-    fun hentForhåndsvisningManueltVarselbrev(@PathVariable behandlingId: UUID,
-                                             malType: Dokumentmalstype,
-                                             fritekst: String): ByteArray {
-        return manueltVarselbrevService.hentForhåndsvisningManueltVarselbrev(behandlingId, malType, fritekst)
+    fun hentForhåndsvisningBrevMedDokumentMal(@PathVariable behandlingId: UUID,
+                                              @RequestBody dto: ForhåndsvisningBrevDto): Ressurs<ByteArray> {
+        if (dto.malType == Dokumentmalstype.INNHENT_DOKUMENTASJON) {
+            return Ressurs.success(innhentDokumentasjonbrevService.hentForhåndsvisningInnhentDokumentasjonBrev(behandlingId,
+                                                                                                               dto.fritekst))
+        } else if (dto.malType == Dokumentmalstype.VARSEL || dto.malType == Dokumentmalstype.KORRIGERT_VARSEL) {
+            return Ressurs.success(manueltVarselbrevService.hentForhåndsvisningManueltVarselbrev(behandlingId,
+                                                                                                 dto.malType,
+                                                                                                 dto.fritekst))
+        } else {
```

Comment:
Hvorfor vil du slå sammen endepunketene. Hvis det er et behov for å slå dem sammen brude logikken flyttes ned i service-klassene.

Reply:
Dette endepunktet skal brukes i Send brev-delen av høgremenyen, og det er betre å kalle berre 1 endepunkt, kor man sender inn kva DokumentMal som skal forhåndsvises